WIP: docs: Migration guide (#257) (#1155)
openbao/openbao: website/content/docs/migration-guide.mdx

---
description: >-

>   This page containes a guide to migrate from Hashicorp Vault to OpenBao

Reviewer: ```suggestion
  This page contains a guide to migrate from HashiCorp Vault to OpenBao
```

---

# Migration guide

> This page contains a guide to migrate a Hashicorp Vault cluster to Openbao.

Reviewer: ```suggestion
This page contains a guide to migrate a HashiCorp Vault cluster to OpenBao.
```

:::info

This guide assumes in-place upgrade, that is, OpenBao will replace the
Vault server process on all nodes. All configuration, including
all endpoints and URLs will remain unchanged.

OpenBao API should be compatible with Vault to the extent that existing
clients should not even register a difference.

Some clients/service might have to be restarted.

:::

## Constraints

 * Vault version 1.14.1 (OSS)
 * OpenBao version 2.2.0
 * Raft storage
 * Shamir unseal (no auto-unseal)

Previous (and future) OpenBao versions will probably work fine. However,
this guide was tested with OpenBao 2.2.0 with Shamir unseal.

> Only the OSS/BUSL version of Vault was tested, not Vault Enterprise.

Reviewer: ```suggestion
Only the (BUSL-licensed) community edition of Vault was tested, not Vault Enterprise.
```

Author: technically 1.14 was still OSS iirc

Reviewer 2: @satoqz @jficz Not to split hairs too much... Vault 1.14.8 and older were OSI licensed, 1.14.9+ were BUSL. Which is... objectively worse? :-D

Reviewer: Ah, gotcha. Then let's not make it complicated, just "community edition" should suffice?

## Limitations

If your Vault version is lower than 1.14.1, upgrade to v1.14.1.

If your Vault has been bootstrapped with Vault version < 1.3,

> you may need to rekey. Vault pre 1.3 used a different Shamir seals

Reviewer: ```suggestion
you may need to rekey. Vault pre 1.3 used a different Shamir seal
```
